### PR TITLE
ci: truncate long k8s namespace names

### PR DESCRIPTION
**Pull request by gcurtis**
They can't be longer than 63 characters.

### CODE REVIEW

File: .github/workflows/previews.yaml [hunk 1]
```diff
@@ -61,6 +61,15 @@ jobs:
...
           curl https://get.jetpack.io -fsSL | bash -s -- -y
+
+          # Make the GitHub branch name safe for Kubernetes namespaces names:
+          #
+          #   * `sed "s/[^0-9a-zA-Z]/-/g"`: replace any invalid characters in
+          #      the branch name with dashes.
+          #   * `tr -s -`: squash repeated consecutive dashes into one.
+          #   * `cut -b -63`: truncate the name to 63 characters.
+          safe_github_ref="$(echo "${GITHUB_REF_NAME}" | sed "s/[^0-9a-zA-Z]/-/g" | tr -s -)"
+          namespace="$(echo "jetpack-io-preview-${safe_github_ref}-ref" | cut -b -63 | tr -s -)"
```
Comment (loreto) [inline .github/workflows/previews.yaml:72]: Is this a preview for the devbox docs?

If yes, could we make the namespace `"$(echo "devbox-docs-preview-${safe_github_ref}"`

**Review by loreto — APPROVED** [review]
LGTM

File: .github/workflows/previews.yaml (hunk 1 above)
Comment (LucilleH) [reply .github/workflows/previews.yaml:72]: Wouldn't truncating this way means it could truncate to a number at the end again?

**Comment by LucilleH** [comment]
Wouldn't truncating this way means it could truncate to a number at the end again?

**Comment by loreto** [comment]
> Wouldn't truncating this way means it could truncate to a number at the end again?

That's a good point. Greg maybe the format should be `devbox-docs-${safe_github_ref}-ref", and you first truncate ``devbox-docs-${safe_github_ref}` to 59 characters and then add `-ref` at the end

**Comment by gcurtis** [comment]
Ugh, good catch! I think everything should work now.

**Pull request merged**